🔨Refactor: 필드 / api 수정

Rosaly-J opened the pull request:
- model, serializers, views, test 코드 수정
- 모델 game_request OnetoOne 필드로 수정
- 자신이 의뢰한 게임에 대해서만 리뷰 작성 가능하게 수정

Rosaly-J committed:
🔨Refactor: 필드 / api 수정

- model, serializers, views, test 코드 수정
- 모델 game_request OnetoOne 필드로 수정
- 자신이 의뢰한 게임에 대해서만 리뷰 작성 가능하게 수정

hwangtate commented on src/reviews/serializers/serializers.py:
```diff
@@ -24,6 +24,16 @@ class Meta:
 
+    def validate(self, data):  # 게임 요청이 사용자와 연결 되어있는지 확인하는 유호성 검사
+        request = self.context.get("request")
+        user = request.user
```
이렇게 user를 가져올 수 있는건가요?

Rosaly-J replied:
request 객체에서 user 속성을 사용해서 가지고 온 뒤에 validate 메서드에서 유효성 검사를 하는 거기 때문에 맞다고 보시면 될 거 같습니다! 혹시 이 부분을 말씀하신 게 아니라면 다시 리플라이 주세요!

hwangtate replied:
포스트맨으로 테스트 할때 user잘 가져와진건가용?

hwangtate commented on src/reviews/serializers/serializers.py:
```diff
@@ -24,6 +24,16 @@ class Meta:
+        game_request = data["game_request"]
+
+        if game_request.user != user:
+            raise serializers.ValidationError("본인이 의뢰한 게임에 대해서만 리뷰를 작성할 수 있습니다.")
```
이 validation에 대한 테스트 코드가 필요 할 것 같아요!

Rosaly-J replied:
확인했습니다! 다시 테스트 코드 짜도록 할게요~

Rosaly-J committed:
🔨Refactor: 필드 / api 수정

- model, serializers, views, test 코드 수정
- 모델 game_request OnetoOne 필드로 수정
- 자신이 의뢰한 게임에 대해서만 리뷰 작성 가능하게 수정
- urls 수정
- 테스트 코드 통과
